refactor(core): remove duplicated code in change_detection_util

mmaterowski opened the pull request:
Removes duplicated code in change_detection_util and reorganizes imports to use utils module.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #45578


## What is the new behavior?

N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
I wasn't able to spin up tests locally, I'm still investigating. Not sure if it's M1 Mac issue. I also get errors regarding
locales import.
eg. _TS2307: Cannot find module '@angular/common/locales/fr' or its corresponding type declarations._

AndrewKushnir commented on packages/core/src/core_private_export.ts:
```diff
@@ -26,8 +25,10 @@ export {allowSanitizationBypassAndThrow as ɵallowSanitizationBypassAndThrow, By
 export {_sanitizeHtml as ɵ_sanitizeHtml} from './sanitization/html_sanitizer';
 export {_sanitizeUrl as ɵ_sanitizeUrl} from './sanitization/url_sanitizer';
 export {coerceToBoolean as ɵcoerceToBoolean} from './util/coercion';
+export {devModeEqual as edevModeEqual} from './util/comparison';
```
```suggestion
export {devModeEqual as ɵdevModeEqual} from './util/comparison';
```

AndrewKushnir replied:
Once this file is updated - the changes from the `goldens/public-api/core/index.md` file would not be needed (the function will be dropped from the golden file).

AndrewKushnir commented on packages/core/src/util/comparison.ts:
```diff
@@ -8,18 +8,18 @@
-export function devModeEqual(a: any, b: any): boolean {
-  const isListLikeIterableA = isListLikeIterable(a);
-  const isListLikeIterableB = isListLikeIterable(b);
+export function devModeEqual(one: any, other: any): boolean {
```
I'd propose rolling this change back (keep current argument names) as it's not related to this change in general.

mmaterowski replied:
Thanks for a feedback @AndrewKushnir.
I applied your suggestions and rolled back this change.
Some tests were failing because of one-letter argument names in function, but I guess now I learned what's the difference between **ɵ** and **e** ^^.

AndrewKushnir reviewed (commented):
@mmaterowski thanks for the cleanup! 👍 

I've just left a few comments, please let me know if you have any questions.

Thank you.

mmaterowski committed:
refactor(core): remove duplicated code in change_detection_util

Removes duplicated code in change_detection_util and reorganizes imports to use utils module.

AndrewKushnir reviewed (approved):
@mmaterowski thanks for addressing the feedback!

The change looks good. We'll run some extra tests in Google's codebase and let you know if there is any additional feedback. If the tests are successful, we'll add the PR to the merge queue.

Thank you.

AndrewKushnir commented:
[Presubmit](http://test/OCL:441248747:BASE:441248785:1649789070571:590581f).

jessicajaniuk commented:
This PR doesn't merge cleanly with `13.3.x`. I've switched to `target: minor` and will land this in master. If you'd like it to land in `13.3.x`, please create a patch PR targeting that branch, and we'll land it separately.

jessicajaniuk commented:
This PR was merged into the repository by commit 663d477cb04da4960a9a6552c556abe73417b95b.
